src/custom/state/enhancedTransactions/updater.tsx

```diff
@@ -0,0 +1,47 @@
+import { sdk } from '@src/custom/utils/blocknative'
+import { useEffect, useMemo } from 'react'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { useActiveWeb3React } from 'hooks/web3'
+import { cancelTransaction, replaceTransaction } from '@src/custom/state/enhancedTransactions/actions'
+
+export default function Updater(): null {
+  const { chainId, library } = useActiveWeb3React()
+  const dispatch = useAppDispatch()
+  const state = useAppSelector((state) => state.transactions)
+
+  const transactions = useMemo(() => (chainId ? state[chainId] ?? {} : {}), [chainId, state])
+
+  useEffect(() => {
+    if (!chainId || !library) return
+
+    const pendingHashes = Object.keys(transactions).filter((hash) => !transactions[hash].receipt)
```

Comment:
Would it make sense to have this inside the useMemo above, so this useEffect only depend on a bunch of hashes? 
I see you don't use `transactions` anywhere else. 

ignore if u feel we will need the transaction info inside the useEffect at a later point